src/HexagonAlignLoads.cpp

```diff
@@ -0,0 +1,243 @@
+#include "HexagonAlignLoads.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+#include "Scope.h"
+#include "Bounds.h"
+#include "ModulusRemainder.h"
+#include "Simplify.h"
+namespace Halide {
+namespace Internal {
+using std::vector;
+
+class HexagonAlignLoads : public IRMutator {
+public:
+    HexagonAlignLoads(Target t, int vs) : target(t), vector_size(vs) {}
+private:
+    Target target;
+    enum AlignCheck { Aligned, Unaligned, NoResult };
+    // size of a vector in bytes.
+    int vector_size;
+    /** Alignment info for Int(32) variables in scope. */
+    Scope<ModulusRemainder> alignment_info;
+    using IRMutator::visit;
+    ModulusRemainder get_alignment_info(Expr e) {
+        return modulus_remainder(e, alignment_info);
+    }
+    int natural_vector_lanes(Type t) {
+        return vector_size / t.bytes();
+    }
+    Expr concat_and_shuffle(Expr vec_a, Expr vec_b, int start, int size) {
+        Type dbl_t = vec_a.type().with_lanes(vec_a.type().lanes() * 2);
+        Expr dbl_vec = Call::make(dbl_t, Call::concat_vectors, { vec_a, vec_b }, Call::PureIntrinsic);
+        std::vector<Expr> args;
+        args.push_back(dbl_vec);
+        for (int i = start; i < start+size; i++) {
+            args.push_back(i);
+        }
+        return Call::make(vec_a.type(), Call::shuffle_vector, args, Call::PureIntrinsic);
```

Comment:
I think this function should just make a vector of Expr indices, and then call the other concat_and_shuffle overload